app/models/health_care_application.rb

```diff
@@ -52,6 +52,26 @@ def process!
     end
   end
 
+  def self.user_icn(form)
+    MVI::Service.new.find_profile_with_attributes(user_attributes(form)).profile.icn
+    # TODO: handle mvi down
+  rescue MVI::Errors::RecordNotFound
+    nil
+  end
```

Comment:
why can't MVI just be fetched through the user model like we do elsewhere? I don't know if I like MVI being invoked from more than one place in the code.

Reply:
I believe the issue is that there might be no user at all so you can use User or the MVI model, without a refactor, as it assumes you're passing in a LO3 user.